includes/Commission/OrderCommission.php

```diff
@@ -0,0 +1,503 @@
+        $gateway_fee = $this->get_dokan_gateway_fee();
+
+        if ( ! empty( $gateway_fee['fee'] ) && self::ADMIN === $gateway_fee['paid_by'] ) {
+            return floatval( $gateway_fee['fee'] );
+        }
+
+        return 0;
+    }
+
+    /**
+     * Get vendor shipping fee.
+     *
+     * @since DOKAN_SINCE
+     *
+     * @return float|int
+     */
+    public function get_vendor_shipping_fee(): float {
+        if ( self::SELLER === dokan()->fees->get_shipping_fee_recipient( $this->order ) ) {
+            return $this->order->get_shipping_total() - $this->get_shipping_refunded();
+        }
+
+        return 0;
+    }
+
+    /**
+     * Get vendor shipping tax fee.
+     *
+     * @since DOKAN_SINCE
+     *
+     * @return float|int
+     */
+    public function get_vendor_shipping_tax_fee(): float {
+        if ( self::SELLER === dokan()->fees->get_shipping_tax_fee_recipient( $this->order ) ) {
+            return ( floatval( $this->order->get_shipping_tax() ) - floatval( $this->get_total_shipping_tax_refunded() ) );
+        }
+
+        return 0;
+    }
+
+    /**
+     * Get vendor tax fee.
+     *
+     * @since DOKAN_SINCE
+     *
+     * @return float|int
+     */
+    public function get_vendor_tax_fee(): float {
+        if ( self::SELLER === dokan()->fees->get_tax_fee_recipient( $this->order->get_id() ) ) {
+            return ( ( floatval( $this->order->get_total_tax() ) - floatval( $this->get_tax_refunded() ) ) - ( floatval( $this->order->get_shipping_tax() ) - floatval( $this->get_total_shipping_tax_refunded() ) ) );
+        }
+
+        return 0;
+    }
+
+    /**
+     * Get vendor gateway fee.
+     *
+     * @since DOKAN_SINCE
+     *
+     * @return float|int
+     */
+    public function get_vendor_earning(): float {
+        return $this->get_vendor_net_earning() + $this->get_total_vendor_fees();
+    }
+
+    /**
+     * Get dokan gateway fee.
+     *
+     * @since DOKAN_SINCE
+     *
+     * @return float|int
+     */
+    public function get_total_admin_fees(): float {
+        return $this->get_admin_shipping_fee() + $this->get_admin_tax_fee() + $this->get_admin_shipping_tax_fee() - $this->get_admin_gateway_fee();
+    }
+
+    /**
+     * Get total vendor fees.
+     *
+     * @since DOKAN_SINCE
+     *
+     * @return float|int
+     */
+    public function get_total_vendor_fees(): float {
+        return $this->get_vendor_shipping_fee() + $this->get_vendor_tax_fee() + $this->get_vendor_shipping_tax_fee() - $this->get_vendor_gateway_fee();
+    }
+
+
+    /**
+     * Get dokan gateway fee.
+     *
+     * @since DOKAN_SINCE
+     *
+     * @return array
+     */
+    private function get_dokan_gateway_fee() {
+        $gateway_fee         = $this->order->get_meta( 'dokan_gateway_fee', true );
+        $gateway_fee_paid_by = $this->order->get_meta( 'dokan_gateway_fee_paid_by', true );
+
+        if ( ! empty( $processing_fee ) && empty( $gateway_fee_paid_by ) ) {
```

Comment:
@Aunshon @mrabbani @kzamanbd should this be `$gateway_fee` instead of `$processing_fee`?

Reply:
@Aunshon @mrabbani @kzamanbd bump

Reply:
https://github.com/getdokan/dokan/pull/3028